utils.py

```diff
@@ -195,3 +196,9 @@ def make_ip_header(src_ip, dest_ip, protocol, data="") -> bytes:
     checksum = calculate_checksum(header)
     # replace the placeholder value with the actual checksum
     return header[:10] + struct.pack("!H", checksum) + header[12:]
+
+def determine_destination_ip_address(url):
+    returned_tuple = urllib.parse.urlparse(url)
+    host_name = returned_tuple.hostname
+    destnation_ip_address = socket.gethostbyname(host_name.decode())
```

Comment:
do we need to do `.local` after like mentioned here: https://piazza.com/class/lcmunpjk5em64t/post/165